scripts/zones/Metalworks/npcs/Medium_Switch.lua

```diff
@@ -0,0 +1,164 @@
+-----------------------------------
+-- Area: Metalworks
+-- NPC:  Medium Switch
+-- Involved in quest: Bait and Switch
+-- !pos -31.176 -1.000 0.043 237
+-----------------------------------
+package.loaded["scripts/zones/Metalworks/TextIDs"] = nil;
+-----------------------------------
+require("scripts/globals/baitandswitch");
+require("scripts/zones/Metalworks/TextIDs");
+require("scripts/globals/settings");
+
+-----------------------------------
+-- onTrade Action
+-----------------------------------
+
+function onTrade(player,npc,trade)
+end; 
+
+-----------------------------------
+-- onTrigger Action
+-----------------------------------
+
+function onTrigger(player,npc)
+    local active = player:getLocalVar("Bait_and_Switch_Quest_Active");
+    local item = CheckOptionForItem(player);
+printf ("Medium Switch - item = %u",item);
+    local increment = player:getLocalVar("CS_Progression_Increment");
+printf ("Medium Switch - increment = %u",increment);
+    local s_table = Switch_Table[player:getLocalVar("Bait_and_Switch_Quest_Order")];
+printf ("Medium Switch - order = %u",player:getLocalVar("Bait_and_Switch_Quest_Order"));
+    local checkSwitch = player:getLocalVar("Switch_Table_Current");
+printf ("Medium Switch - checkSwitch = %u",checkSwitch);
+    local paramEight = 0;
+printf ("Medium Switch - Lead Guardsman = %u",player:getLocalVar("Lead_Guardsman"));
+        if (item == 5) then
+            if (player:getLocalVar("Lead_Guardsman") == BnS_REPLICA) then
+                if (s_table.ord[checkSwitch] == 2) then
+                    paramEight = 3;
+                else
+                    paramEight = 4;
+                end
+            end
+        elseif (item == 4) then
+            if (CheckSnares(player)[7] <= 3) then
+                if (CheckSnares(player)[4] == 1) then
+                    paramEight = 3;
+                end
+            elseif (CheckSnares(player)[7] > 3) then
+                if (CheckSnares(player)[4] == 1) then
+                    paramEight = 4;
+                end
+            end
+        end
+    local CSProgression = CheckBaitProgress(player)[item] + increment;
+printf ("Medium Switch - CS Progression = %u",CSProgression);
+        if (CSProgression > 3) then
+            CSProgression = 0;
+        end
+    local caughtParam = 0;
+    local caught = player:getLocalVar("Bait_and_Switch_Caught");
+printf ("Medium Switch - caught = %u",caught);
+        if (caught == 2) then
+            caughtParam = 1;
+        end
+    local timeCheck = player:getLocalVar("Bait_and_Switch_Time_Limit");
+
+    if (os.time() > timeCheck and item == 6) then
+        player:messageSpecial(BAIT_AND_SWITCH_BASE - 380);
+        player:resetLocalVars();
+    elseif (active == 1) then
+        player:startEvent(0x038d,0,caughtParam,item,0,0,CSProgression,0,paramEight);
+    end
```

Comment:
most of this could also probably fit in the global

Reply:
i dont know about this being global.  it starts a cutscene event... im not sure exactly how you would want me to implement that.

Reply:
there we go... put a lot of stuff in global